Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.m

```diff
@@ -34,48 +34,7 @@ - (NSDictionary *)vok_dictionaryRepresentationRespectingKeyPaths
 
 + (NSString *)vok_entityName
 {
-    static char AssociatedObjectKey;  // The key for the runtime-association.
-    
-    // Get the associated value.
-    NSString *vok_entityName = objc_getAssociatedObject(self, &AssociatedObjectKey);
-    
-    // If we didn't find an associated entity name, determine the entity name.
-    if (!vok_entityName) {
-        // If we already have an entityName class method (e.g., MOGenerator-generated subclasses), use it.
-        // (Note that we have to cast self (a Class) to id to use NSObject's dynamic-selector methods, even though they work.)
-        if ([(id)self respondsToSelector:@selector(entityName)]) {
-            vok_entityName = [(id)self performSelector:@selector(entityName)];
-        }
-        if (!vok_entityName) {
-            // On OS X, NSObject has a private class method called entityName but it may return nil.
-            // https://github.com/rentzsch/mogenerator/issues/196
-            
-            // Since we don't have an entityName class method (or it didn't return a result),
-            // look up the entity name in the managed object model.
-            NSManagedObjectModel *model = [[VOKCoreDataManager sharedInstance] managedObjectModel];
-            
-            // Start with the class we are...
-            Class workingClass = self;
-            do {
-                NSString *workingClassName = NSStringFromClass(workingClass);
-                // ... check for a matching entity in the model...
-                for (NSEntityDescription *description in model.entities) {
-                    if ([workingClassName isEqualToString:description.managedObjectClassName]) {
-                        vok_entityName = description.name;
-                        break;
-                    }
-                }
-                // ... and walk up the superclass chain...
-                workingClass = [workingClass superclass];
-                // ... until we get Nil or find a matching entity (as long as we have a superclass to test and haven't found the entity name).
-            } while (workingClass && !vok_entityName);
-        }
-        NSAssert(vok_entityName, @"no entity found that uses %@ as its class", NSStringFromClass(self));
-        // Save the determined entity name as an associated value.
-        objc_setAssociatedObject(self, &AssociatedObjectKey, vok_entityName, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
-    }
-    
-    return vok_entityName;
+    return [[VOKCoreDataManager sharedInstance] entityNameForClass:self];
```

Comment:
That expectation should be prominently documented in the header for this category.